lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -106,10 +107,18 @@ def device_change attributes
         def boot_options attributes
           # NOTE: you must be using vsphere_rev 5.0 or greater to set boot_order
           # e.g. Fog::Compute.new(provider: "vsphere", vsphere_rev: "5.5", etc)
-          return unless @vsphere_rev.to_f >= 5
-          RbVmomi::VIM::VirtualMachineBootOptions.new(
-            :bootOrder => boot_order(attributes)
-          )
+          options = {}
+          if @vsphere_rev.to_f >= 5 and attributes[:boot_order]
+            options[:bootOrder] = boot_order(attributes)
+          end
+
+          # Set attributes[:boot_retry] to a delay in miliseconds to enable boot retries
+          if attributes[:boot_retry]
+            options[:bootRetryEnabled] = true
```

Comment:
The limitation is that right now the attributes of the model are pulled in by the `convert_vm_view_to_attr_hash` function in compute.rb. This currently maps properties one-to-one with model attributes.

The two attributes `:boot_order` (this was already present) and `:boot_retry` (my parameter) construct more complex properties by expanding the single value given in Ruby to a more complex multi-value SOAP type. When retrieving a VM, it is this type that will come back and so if we wanted to store these values in the model, we would have to reverse the process of creating these types at VM creation time.

Since these are boot-related options, people are unlikely to need them beyond the creation of the VM. I'm assuming that's the reason the `:boot_order` was never added to the model and why I think perhaps it's not worth adding `:boot_retry` to the model either.